AssertionError: mkdoc.py does NOT exist

PR #9510 broke bazel_drake_external workflows:

```console
ERROR: /home/ubuntu/.tmpfs/bazel_output_base/external/drake/bindings/pydrake/BUILD.bazel:33:1: Couldn't build file external/drake/bindings/pydrake/documentation_pybind.h: error executing shell command: 'bazel-out/host/bin/external/drake/tools/workspace/pybind11/mkdoc $@ > bazel-out/k8-opt/genfiles/external/drake/bindings/pydrake/documentation_pybind.h' failed (Exit 1)
Traceback (most recent call last):
  File "/home/ubuntu/.tmpfs/bazel_output_base/sandbox/linux-sandbox/1/execroot/enabling_anzu/bazel-out/host/bin/external/drake/tools/workspace/pybind11/mkdoc.runfiles/drake/tools/workspace/pybind11/run_mkdoc_with_python3.py", line 23, in <module>
    mkdoc_path = resolve_path('third_party/com_github_pybind_pybind11/mkdoc.py')
  File "/home/ubuntu/.tmpfs/bazel_output_base/sandbox/linux-sandbox/1/execroot/enabling_anzu/bazel-out/host/bin/external/drake/tools/workspace/pybind11/mkdoc.runfiles/drake/tools/workspace/pybind11/run_mkdoc_with_python3.py", line 7, in resolve_path
    assert os.path.exists(abspath), 'Path {} does NOT exist.'.format(abspath)
AssertionError: Path /home/ubuntu/.tmpfs/bazel_output_base/sandbox/linux-sandbox/1/execroot/enabling_anzu/bazel-out/host/bin/external/drake/tools/workspace/pybind11/mkdoc.runfiles/enabling_anzu/third_party/com_github_pybind_pybind11/mkdoc.py does NOT exist.
```

--- a/tools/workspace/pybind11/run_mkdoc_with_python3.py
+++ b/tools/workspace/pybind11/run_mkdoc_with_python3.py
@@ -20,7 +20,12 @@ if sys.platform.startswith('linux'):
 
 os.environ['LANG'] = 'en_US.UTF-8'
 
-mkdoc_path = resolve_path('third_party/com_github_pybind_pybind11/mkdoc.py')
+mkdoc_relpath = 'third_party/com_github_pybind_pybind11/mkdoc.py'
+
+if os.path.exists(os.path.join(runfiles_dir, 'external/drake')):
+    mkdoc_relpath = os.path.join('external/drake', mkdoc_relpath)
+
+mkdoc_path = resolve_path(mkdoc_relpath)
 python3_path = resolve_path('external/python3/python3')
 argv = [python3_path, mkdoc_path] + sys.argv[1:]
 os.execv(argv[0], argv)

